invoice.go

```diff
@@ -6,125 +6,131 @@ import "encoding/json"
+	ApplicationFee      *int64   `form:"application_fee"`
+	Billing             *string  `form:"billing"`
+	Closed              *bool    `form:"closed"`
+	Customer            *string  `form:"customer"`
+	DaysUntilDue        *int64   `form:"days_until_due"`
+	Description         *string  `form:"description"`
+	DueDate             *int64   `form:"due_date"`
+	Forgiven            *bool    `form:"forgiven"`
+	Paid                *bool    `form:"paid"`
+	StatementDescriptor *string  `form:"statement_descriptor"`
+	Subscription        *string  `form:"subscription"`
+	TaxPercent          *float64 `form:"tax_percent"`
 
 	// These are all for exclusive use by GetNext.
 
-	SubItems         []*SubItemsParams `form:"subscription_items,indexed"`
-	SubNoProrate     bool              `form:"subscription_prorate,invert"`
-	SubPlan          string            `form:"subscription_plan"`
-	SubProrationDate int64             `form:"subscription_proration_date"`
-	SubQuantity      uint64            `form:"subscription_quantity"`
-	SubQuantityZero  bool              `form:"subscription_quantity,zero"`
-	SubTrialEnd      int64             `form:"subscription_trial_end"`
+	SubscriptionItems         []*SubscriptionItemsParams `form:"subscription_items,indexed"`
+	SubscriptionPlan          *string                    `form:"subscription_plan"`
+	SubscriptionProrate       *bool                      `form:"subscription_prorate"`
+	SubscriptionProrationDate *int64                     `form:"subscription_proration_date"`
+	SubscriptionQuantity      *int64                     `form:"subscription_quantity"`
+	SubscriptionTrialEnd      *int64                     `form:"subscription_trial_end"`
 }
 
 // InvoiceListParams is the set of parameters that can be used when listing invoices.
 // For more details see https://stripe.com/docs/api#list_customer_invoices.
 type InvoiceListParams struct {
-	ListParams `form:"*"`
-	Billing    InvoiceBilling    `form:"billing"`
-	Customer   string            `form:"customer"`
-	Date       int64             `form:"date"`
-	DateRange  *RangeQueryParams `form:"date"`
-	DueDate    int64             `form:"due_date"`
-	Sub        string            `form:"subscription"`
+	ListParams   `form:"*"`
+	Billing      *string           `form:"billing"`
+	Customer     *string           `form:"customer"`
+	Date         *int64            `form:"date"`
+	DateRange    *RangeQueryParams `form:"date"`
+	DueDate      *int64            `form:"due_date"`
+	Subscription *string           `form:"subscription"`
 }
 
 // InvoiceLineListParams is the set of parameters that can be used when listing invoice line items.
 // For more details see https://stripe.com/docs/api#invoice_lines.
 type InvoiceLineListParams struct {
 	ListParams `form:"*"`
 
-	Customer string `form:"customer"`
+	Customer *string `form:"customer"`
 
 	// ID is the invoice ID to list invoice lines for.
-	ID string `form:"-"` // Goes in the URL
+	ID *string `form:"-"` // Goes in the URL
```

Comment:
@brandur-stripe Should we remove this and change the prototype of the method to take the id as a parameter instead?

Reply:
Good call ... that would probably be the better interface. If we did it though, we'd probably want to change other places that get a URL parameter based a `Params` struct though — application fee refund's `New` for example.

Looking around, there are quite a few of these (I used `ag '/%v/'` as a heuristic to find them), and changing them all might be a project of its own. In the spirit of changing less, I could see just leaving them as is, but I don't feel super strongly either way. Thoughts?

Reply:
I think there's a difference when the parameter is just the ID versus when the parameter is dependent on the type of resource like `/v1/customers/xxx/sources/yyy` insread of `/v1/recipients/xxx/sources/yyy`.
But I'm fine leaving as is for now

Reply:
Yeah, let's just leave it for the time being then.